lib/client.js

```diff
@@ -73,7 +73,8 @@ class Client {
   constructor(room, socket, rtoken) {
     const {remoteAddress, remotePort} = socket.request.connection;
     this.socket = socket;
-    this.ip = socket.request.headers['x-real-ip'] || remoteAddress;
+    let xForwardFor = CONFIG.get("useXForwardedFor") ? socket.request.headers["x-forwarded-for"] : null;
+    this.ip = xForwardFor ? xForwardFor.split(",")[0] : remoteAddress;
```

Comment:
two tertiaries and a null gibs headaches :*/
please for the love of god, don't. There is nothing wrong with an if or two...

Reply:
Wuuuuut.   Trinary is akin to the holy trinity.